### PR TITLE
Inital RPC Layer w/ Example

### PR DESCRIPTION
**Pull request by JackyChiu**
# What

This PR adds the initial RPC layer #27. This is demonstrated with a simple calculator server running locally. 
  
# How
The client utilizes the service registry to find the node to connect to. It currently uses a naive algorithm that picks the first node found in the list. We should do something similar to etcd strategy and connect to a random node in the list. 

To test the client a refactor of the registry was needed to decouple the implementation. 

Run the example code with 
```go
cd examples/calculator
./run
```

### CODE REVIEW

File: example/calculator/client/client.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/edegens/ptype/cluster"
+	"github.com/edegens/ptype/example/calculator"
+)
+
+func main() {
+	cfg, err := cluster.ConfigFromFile(os.Getenv("CONFIG"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	capnslog.SetGlobalLogLevel(capnslog.ERROR)
```
Comment (JackyChiu) [inline example/calculator/client/client.go]: Temp solution until we can upgrade to etcd v3.4 and use their log level configs

File: cluster/registry.go [hunk 2]
```diff
@@ -13,16 +13,21 @@ import (
...
+	Services(ctx context.Context) (map[string][]Node, error)
+}
+
 type Node struct {
 	Address string `json:"address"`
 	Port    int    `json:"port"`
 }
 
-type ServiceRegistry struct {
+type serviceRegistry struct {
```
Comment (JackyChiu) [inline cluster/registry.go]: do you guys think it's necessary to name this `etcdRegistry` instead? since we can support different stores now?

Comment (bkzhang) [reply cluster/registry.go]: I'm fine with etcdRegistry but why did you make this lower case?

Comment (JackyChiu) [reply cluster/registry.go]: I didn't see a need to expose the underlying implimentation of the registry with the introduction of the interface

Comment (bkzhang) [reply cluster/registry.go]: right mb, i completely forgot about how golang uses caps for private/public

File: cluster/rpc.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		return nil, err
+	}
+
+	nodes, ok := serv[serviceName]
+	if !ok || len(nodes) == 0 {
+		return nil, fmt.Errorf("failed to find service %v in the registry", serviceName)
+	}
+
+	// naively pick first element; algorithm for choosing node can be improved
+	dailAddr := fmt.Sprintf("%v:%v", nodes[0].Address, nodes[0].Port)
```
Comment (bkzhang) [inline cluster/rpc.go]: dial?

**Pull request merged**